File: platforms/macos/Aether/Sources/MultiTurn/Models/ContentDisplayState.swift
//
//  ContentDisplayState.swift
//  Aether
//
//  Display state for unified conversation window content area.
//

import Foundation

// MARK: - ContentDisplayState

/// Mutually exclusive display states for the content area above input
enum ContentDisplayState: Equatable {
    /// No conversation, no commands - initial state
    case empty

    /// Showing conversation history
    case conversation

    /// Showing command or topic list
    case commandList(prefix: String)  // "/" for commands, "//" for topics

    /// Check if showing command list
    var isShowingCommandList: Bool {
        if case .commandList = self {
            return true
        }
        return false
    }

    /// Check if showing topic list (// prefix)
    var isShowingTopicList: Bool {
        if case .commandList(let prefix) = self {
            return prefix == "//"
        }
        return false
    }

    /// Check if showing command list (/ prefix, not //)
    var isShowingCommands: Bool {
        if case .commandList(let prefix) = self {
            return prefix == "/"
        }
        return false
    }
}

File: platforms/tauri/src/windows/halo/types/ContentDisplayState.ts
// Content display states - mutually exclusive
export type ContentDisplayState =
  | { type: 'empty' }                           // Initial: only input box
  | { type: 'conversation' }                    // Show conversation history
  | { type: 'commandList'; prefix: string }     // "/" commands
  | { type: 'topicList'; prefix: string };      // "//" topics

// Helper functions
export function isShowingPanel(state: ContentDisplayState): boolean {
  return state.type !== 'empty';
}

export function isShowingCommandList(state: ContentDisplayState): boolean {
  return state.type === 'commandList';
}

export function isShowingTopicList(state: ContentDisplayState): boolean {
  return state.type === 'topicList';
}
